opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
+ * amount of connection information that identifies the endpoints
+ * endpoint.
+ */
+
```

Comment:
This function (and the corresponding send) should now be updated to handle the timeout flags. Reading through the socket(7) man page, the SO_RCVTIMEO and SO_SNDTIMEO would force the return with EAGAIN, EWOULDBLOCK or EINPROGRESS. Currently these return values are not correctly handled (most of the are ignored).

The comment states that the socket is in non-blocking mode. If this was true then SO_RCVTIMEO would have no impact, as the recv would return EWOULDBLOCK as soon as no data is pending on the socket and the timeout would never trigger. Luckily the comment is not accurate as when this mca_btl_tcp_recv_blocking is called, from mca_btl_tcp_component_recv_handler (thus from the accept callback), the socket is still in blocking mode (as we use accept). That being said my comment just above should be addressed.